pkg/route/ingress/ingress.go

```diff
@@ -415,6 +416,18 @@ func (c *Controller) sync(key queueKey) error {
 		old = append(old, route)
 	}
 
+	// In case the annotation does not contain a valid value, the reconciliation
+	// should continue, so we don't return the error, but instead register an
+	// event of InvalidAnnotationValue.
+	propagateLabels, err := shouldPropagateLabelsToRoute(ingress.Annotations)
+	if err != nil {
+		c.eventRecorder.Eventf(&corev1.ObjectReference{
+			Kind:      "Ingress",
+			Namespace: key.namespace,
+			Name:      key.name,
+		}, corev1.EventTypeNormal, "InvalidAnnotationValue", "Invalid value on annotation %q", routecontroller.PropagateIngressLabelFlag)
```

Comment:
```suggestion
		}, corev1.EventTypeNormal, "InvalidAnnotationValue", "Invalid value on annotation %q due to: %q", routecontroller.PropagateIngressLabelFlag, err)
```

It could be useful to add the error in the event to help diagnostic and debugging